### DISCUSSION TITLE
Postgresql TypeError: Do not know how to serialize a BigInt

### DISCUSSION THREAD

**Discussion by Daniek3d** [prisma/prisma#5737] [Q&A] [answered]
Hello, when i select a table info with fiels with bitint type it dont working, and sende me its error.

TypeError: Do not know how to serialize a BigInt
    at JSON.stringify (<anonymous>)

Thanks

**Comment by ryands17** [comment]
@Daniek3d 👋 

`JSON.stringify` cannot directly stringify a `bigint`. To do this you need to perform it in the following manner:

```ts
JSON.stringify(data, (key, value) =>
            typeof value === 'bigint'
                ? value.toString()
                : value
)
```

**Reply by Daniek3d** [reply]
where write this code ?

**Reply by ryands17** [reply]
Wherever you have `Json.stringify` called in your code.

**Reply by Daniek3d** [reply]
Sorry but i dont call this function in my code. this error its when read data from prisma schema.

My code:
```
const data = await this.prisma.dig.findMany({
                select:{
                    record: true,
                    orderno: true,
                    key: true,
                   time: true
                },
                where:{
                    
                }
            })
```;

Mi Model:
model digitales {
  record               Int      @id
orderno:             Int
time:                  BigInt?
........


Thanks

**Reply by ryands17** [reply]
It seems you are on an older version of Prisma. I would suggest installing the latest version of Prisma `2.17.0` and then trying again.

```
npm i prisma -D
```

This will install the latest version and I tried that on a sample model which works fine with `BigInt`.

![image](https://user-images.githubusercontent.com/19697099/108547262-450bbc00-7310-11eb-933a-ba37452c8b4d.png)

**Reply by pedrosc94** [reply]
@Daniek3d Found any solution?

**Reply by Daniek3d** [reply, marked as answer]
This workin in the new ralease.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
